graphrag_sdk/models/litellm.py

```diff
@@ -0,0 +1,283 @@
+import os
+import logging
+from ollama import Client
+from typing import Optional
+from litellm import completion
+from .model import (
+    OutputMethod,
+    GenerativeModel,
+    GenerativeModelConfig,
+    GenerationResponse,
+    FinishReason,
+    GenerativeModelChatSession,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO) 
+
+class LiteModel(GenerativeModel):
+    """
+    A generative model that interfaces with the LiteLLM for chat completions.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        generation_config: Optional[GenerativeModelConfig] = None,
+        system_instruction: Optional[str] = None,
+    ):
+        """
+        Initialize the LiteModel with the required parameters.
+
+        Args:
+            model_name (str): The name and the provider for the LiteLLM client.
+            generation_config (Optional[GenerativeModelConfig]): Configuration settings for generation.
+            system_instruction (Optional[str]): Instruction to guide the model.
+        """
+
+        self.model = model_name
+        if "/" in model_name:
+            self.provider = model_name.split("/")[0]
+            self.model_name = model_name.split("/")[1]
+        else:
+            self.provider = "openai"
+            self.model_name = model_name
```

Comment:
A bit weird to assume `openai` in this case.

Reply:
OpenAI is the only vendor that not need to specify the provider:
"azure/gpt-4o" -> azure
"gpt-4o"/"openai/gpt-4o" -> openai